### DISCUSSION TITLE
error when updating 1.30.11 -> 2.x.x: sqlstate[01000]: warning: 1265 truncated fro column 'company' at row 43

### DISCUSSION THREAD

**Discussion by edvc2331** [kimai/kimai#4455] [On-Premise: Installation & Updates] [answered]
hi,

i run an ubuntu 20.04 here with php 7.4 running kimai 1.30.11

first i updated to php 8.2 - worked. then i tried to update to kimai 2.4.1

removed the plugins and cleared the cache following the upgrade guidelines

i get this error after bin/console kimai:update:

Kimai updates running ...
=========================

[notice] Migrating up to DoctrineMigrations\Version20230819090536
[error] Migration DoctrineMigrations\Version20230126002049 failed during Execution. Error: "An exception occurred while executing a query: SQLSTATE[01000]: Warning: 1265 Data truncated for column 'company' at row 42"

 [ERROR] An exception occurred while executing a query: SQLSTATE[01000]: Warning: 1265 Data truncated for column
         'company' at row 42

any idea? maybe an umlaut somewhere? äöü?

thank you!

**Comment by kevinpapst** [comment, marked as answer]
Out of thousands of installations you are the second person that runs into this issue 🤷  Sorry!

The migration reduces the size of some columns and usually no one used the full length (which was accidentally too long before):
https://github.com/kimai/kimai/blob/main/migrations/Version20230126002049.php#L35

Check your customer table. 
Seems you have at least one row where the `company` name is unusually long. 

Do you speak SQL to fix that yourself?

**Reply by edvc2331** [reply]
thank you for your quick response (as always). i used these commands to find company names longer than 100 chars

mysql -uroot kimai2 -p
select company from kimai2_customers WHERE CHAR_LENGTH(company) > 100;
quit

solved!
